test/fr_fastq_diff_test.c

```
@@ -0,0 +1,278 @@
+    int fd = open(path, O_RDONLY);
+    assert(fd >= 0);
+    const char *err = NULL;
+    fast_reader_t *fr = fast_reader_dopen(fd, &err);
+    assert(fr != NULL);
+    kseq_t *ks = kseq_init(fr);
+
+    int cap = 16, n = 0;
+    rec_t *recs = (rec_t *)malloc(sizeof(rec_t) * cap);
+    assert(recs != NULL);
+    while (kseq_read(ks) >= 0) {
+        if (n == cap) { cap *= 2; recs = (rec_t *)realloc(recs, sizeof(rec_t) * cap); assert(recs); }
```

Cppcheck (2.21.0)

[error] Common realloc mistake

(memleakOnRealloc)